File: CEopt-1.0/MainCEoptExample1.m
clc; clear; close all;

disp(' ------------------- ')
disp(' MainCEoptExample1.m ')
disp(' ------------------- ')

% objective function
F = @(x) -0.8*exp(-(x-2).^2) - 0.5*exp(-(x+2).^2)+1;

% bound for design variables
lb = -5;
ub =  5;

% initialize mean and std. dev. vectors
mu0    = (ub+lb)/2;
sigma0 = (ub-lb)/6;

% CE optimizer
tic
[Xopt,Fopt,ExitFlag,CEobj] = CEopt(F,mu0,sigma0,lb,ub)
toc

File: CEopt-1.0-Python/MainCEoptExample1.py
import time
import numpy as np
from CEopt import CEopt  # Assumes CEopt.py is in the same directory

print(' ------------------- ')
print(' MainCEoptExample1.py ')
print(' ------------------- ')

# Objective function
# -----------------------------------------------------------------
def F(x):
    # x can be a scalar or an array; operations are elementwise.
    return -0.8 * np.exp(-(x - 2)**2) - 0.5 * np.exp(-(x + 2)**2) + 1
# -----------------------------------------------------------------

# Bound for design variables
lb = -5
ub =  5

# Initialize mean and standard deviation vectors.
mu0    = (ub + lb) / 2.0
sigma0 = (ub - lb) / 6.0

# Run the CE optimizer
start_time = time.time()
Xopt, Fopt, ExitFlag, CEobj = CEopt(F, mu0, sigma0, lb, ub)
elapsed_time = time.time() - start_time

print("\nXopt =", Xopt)
print("Fopt =", Fopt)
print("ExitFlag =", ExitFlag)
#print("CEobj =", CEobj)
print("Elapsed time: {:.4f} seconds".format(elapsed_time))
